File: week1/day5/algo-calculate-mode/js/calculateMode.js
exports.calculateMode = function(list) {

    let dict = {}
    let result = []

    for (i in list) {
        if (list[i] in dict) {
            dict[list[i]] += 1
        } else {
            dict[list[i]] = 1
        }
    }

    // console.log(dict)

    let top_count = 0

    for (key in dict) {
        if (dict[key] > top_count) {
            top_count = dict[key]
        }
    }

    for (key in dict) {
        if (dict[key] === top_count) {
            result.push(key)
        }
    }

    

    console.log(result)

    return result

}

File: week1/day5/algo-calculate-mode/python/calculate_mode.py
from itertools import count


def calculate_mode(list):
    count_dict = {}

    result = []

    for char in list:
        if char in count_dict:
            count_dict[char] += 1

        else:
            count_dict[char] = 1

    top_count = 0

    # print(count_dict)

    for value in count_dict:
        if count_dict[value] > top_count:
            top_count = count_dict[value]

    for value in count_dict:
        if count_dict[value] == top_count:
            result.append(value)

    return result